src/java.desktop/windows/classes/sun/awt/windows/ThemeReader.java

```diff
@@ -107,14 +107,23 @@ private static Long openThemeImpl(String widget, int dpi) {
            // We're using the syntax "subAppName::controlName" here, as used by msstyles.
            // See documentation for SetWindowTheme on MSDN.
            setWindowTheme(widget.substring(0, i));
-           theme = openTheme(widget.substring(i + 2), dpi);
+           theme = getOpenThemeValue(widget.substring(i + 2), dpi);
            setWindowTheme(null);
        } else {
-           theme = openTheme(widget, dpi);
+           theme = getOpenThemeValue(widget, dpi);
        }
        return theme;
     }
 
+    private static Long getOpenThemeValue(String widget, int dpi) {
+        Long theme;
+        theme = openTheme(widget, dpi);
+        if (theme == 0) {
+            theme = openTheme(widget, defaultDPI);
+        }
+        return theme;
+    }
```

Comment:
```suggestion
    private static long getOpenThemeValue(String widget, int dpi) {
        long theme = openTheme(widget, dpi);
        if (theme == 0) {
            theme = openTheme(widget, defaultDPI);
        }
        return theme;
    }
```
Yes, that's what I meant. Let's use the primitive `long`.